web/search.dart

```diff
@@ -251,21 +292,71 @@ void initializeSearch(
   }
 
   void showSuggestions() {
-    if (presentation.hasChildNodes()) {
+    if (searchResults.hasChildNodes()) {
       listBox.style.display = 'block';
       listBox.setAttribute('aria-expanded', 'true');
     }
   }
 
+  // Function that iterates through the categoriesMap and appends it to the given HTML element
+  // The received Element is the one used for search results page or dropdown list in the search bar
+  void iterateCategoriesMap(Element e) {
+    categoriesMap.forEach((keys, values) {
+      var value = categoriesMap[keys];
+      e.append(value!);
+    });
+  }
+
+  // Function that creates the content displayed in the main-content element.
+  void searchResultPage(String input) {
```

Comment:
I think this function name would read better with a verb in front, like 'showSearchResultPage'. Although I can see the appeal of a more declarative API, with functions named as the nouns they produce... I just don't think its a convention we have.

Reply:
That is a good suggestion, thank you